metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -686,7 +687,15 @@ public Long apply(List<ApiMessageAndVersion> records) {
                             int i = 1;
                             for (ApiMessageAndVersion message : records) {
                                 try {
-                                    replay(message.message(), Optional.empty(), prevEndOffset + records.size());
```

Comment:
Here we were using `prevEndOffset + records.size()` to make it consistent with the code in `handleCommit` and `handleSnapshot`.